base/reflection.jl

```diff
@@ -984,6 +985,11 @@ function code_typed(@nospecialize(f), @nospecialize(types=Tuple);
         debuginfo == :none && remove_linenums!(code)
         push!(asts, code => ty)
     end
+    if trace && !isempty(params.trace_buffer)
+        for entry in params.trace_buffer
+            @info entry
```

Comment:
code_typed shouldn't print—and this seems more like a code_warntype feature anyways?

Reply:
We could separate this functionality out and expect users to pass in the right `CustomParams` or alternatively a separate macro `@explain`/`
@WHY?!!` that does this.